feat [IDP-1510] : Added the tutorial to add links in software components (#3889)
harness/developer-hub: tutorials/internal-developer-portal/add-links-in-components.md

---
title: How to add Links in Software Components
description: Steps to add various links associated with the component.
sidebar_position: 4
---

## Introduction

> In this tutorial we will explore on how to add links associated with the Software Components which includes user profiles, groups, dashboards, support pages, and more. Each link can be accompanied by an icon for visual representation.

Reviewer: ```suggestion
In this tutorial we will explore how to add links associated with the Software Components which could be dashboards, support pages, runbooks, email groups, Slack channels, etc. Each link is an arbitrary name and URL pair, which can be accompanied by an icon for visual representation.
```

Once you register a Software Component in the Catalog this is a sample of how your overview section would look based on your IDP yaml. 

![](./static/link-card.png)

> In the depicted image, you'll notice a **Links Card** that serves as a navigational tool. These links can be anything from the following suggested categories:

Reviewer: ```suggestion
In the depicted image, you'll notice a **Links Card** that serves as a navigational tool. These links can be anything. Here are some examples -
```

- User and Group Management
- Infrastructure and Cloud
- Operational Dashboards
- Support and Help
- General Web and Information
- Notifications and Alerts

and many more which would function as a quick bookmark for the developers corresponding to the software component. 

Follow the steps below to effectively integrate such links into your software components in the catalog.

## Steps to add Links
 
 1. Click on the edit icon on the overview page, which will redirect you to the IDP yaml.
 
 ![](./static/edit-icon.png)

 2. Now add **links** to the IDP yaml as shown in the example below.
    
<details>
<summary>Link Attributes</summary>

- **url:** This is the actual web address or URL that the link points to. It's the destination users will be directed to when they click on the link.

- **title:** This serves as the display name or label for the link. It gives users a brief idea of what to expect when they click on the link or where it might lead them.

> - **icon:** An optional attribute, you can select from a category of supported icon that provides a visual representation or hint about the nature or category of the link. 

Reviewer: Is there a place where we can see all available links? I suppose this is MUI icons. cc @jenil-harness

Reviewer 2: We are going to replace this with harness icons. I will share more details once the implementation is done
